docker-compose/tomcat-manager/docker-compose.yml

```diff
@@ -0,0 +1,241 @@
+        -Dapiman.es.timeout=10000
+        -Dapiman.es.username=${ELASTICSEARCH_USERNAME}
+        -Dapiman.es.password=${ELASTICSEARCH_PASSWORD}
+        -Dapiman.truststore-password=${TRUSTSTORE_KEYSTORE_PASSWORD}
+        -Dapiman.public-endpoint=${ENDPOINT}
+        -Dkeycloak.protocol=${KEYCLOAK_PROTOCOL}
+        -Dkeycloak.host=${KEYCLOAK_HOST}
+        -Dkeycloak.port=${KEYCLOAK_PORT}
+        -Dkeycloak.realm=${KEYCLOAK_REALM}
+        -DallowSelfSigned=${SELF_SIGNED}
+        -DallowAnyHost=${SELF_SIGNED}
+        -Dkeycloak.realm-public-key=${KEYCLOAK_REALM_PUBLIC_KEY}
+        -Dkeycloak.gateway.secret=${KEYCLOAK_GATEWAY_SECRET}
+        -DcachingPolicy.maxCacheSize=${MAX_CACHE_SIZE_IN_MB}
+        -Dapiman.gateway-backend-keystore=${KEYSTORE_TRUSTSTORE_PATH:-}${GATEWAY_BACKEND_KEYSTORE:-/usr/src/apiman/apiman-distro-vertx/certs/apiman.jks}
+        -Dapiman.gateway-backend-truststore=${KEYSTORE_TRUSTSTORE_PATH:-}${GATEWAY_BACKEND_TRUSTSTORE:-/usr/src/apiman/apiman-distro-vertx/certs/apiman.jks}
+        -Dapiman.gateway-client-keystore=${KEYSTORE_TRUSTSTORE_PATH:-}${GATEWAY_CLIENT_KEYSTORE:-/usr/src/apiman/apiman-distro-vertx/certs/apiman.jks}
+        -Dapiman.gateway-client-truststore=${KEYSTORE_TRUSTSTORE_PATH:-}${GATEWAY_CLIENT_TRUSTSTORE:-/usr/src/apiman/apiman-distro-vertx/certs/apiman.jks}
+        -Dapiman.gateway-es-keystore=${KEYSTORE_TRUSTSTORE_PATH:-}${GATEWAY_ES_KEYSTORE:-/usr/src/apiman/apiman-distro-vertx/certs/apiman.jks}
+        -Dapiman.gateway-es-truststore=${KEYSTORE_TRUSTSTORE_PATH:-}${GATEWAY_ES_TRUSTSTORE:-/usr/src/apiman/apiman-distro-vertx/certs/apiman.jks}
+        -Dapiman.gateway-keycloak-keystore=${KEYSTORE_TRUSTSTORE_PATH:-}${GATEWAY_KEYCLOAK_KEYSTORE:-/usr/src/apiman/apiman-distro-vertx/certs/apiman.jks}
+        -Dapiman.gateway-keycloak-truststore=${KEYSTORE_TRUSTSTORE_PATH:-}${GATEWAY_KEYCLOAK_TRUSTSTORE:-/usr/src/apiman/apiman-distro-vertx/certs/apiman.jks}
+        -Dapiman.gateway-allowedProtocols=${TLS_ALLOWED_PROTOCOLS:-}
+        -Dapiman.public-endpoint=${ENDPOINT}:80/gateway
+#    ports:
+#      - '${GATEWAY_PORT}:${GATEWAY_PORT}'
+#      - '${GATEWAY_MANAGEMENT_PORT}:8081'
+    volumes:
+      - type: volume
+        source: gateway_logs
+        target: /usr/src/apiman/apiman-distro-vertx/logs
+    logging:
+      driver: "json-file"
+      options:
+        max-size: "10m"
+        max-file: "10"
+        mode: non-blocking
+    labels:
+#      - traefik.http.routers.gateway-router.tls=true
+      - traefik.http.routers.gateway-router.entrypoints=http
+      - traefik.http.routers.gateway-router.rule=PathPrefix(`/gateway`)
+      - traefik.http.routers.gateway-router.service=gateway-service
+      - traefik.http.routers.gateway-router.middlewares=stripprefix
+      - traefik.http.middlewares.stripprefix.stripprefix.prefixes=/gateway
+      - traefik.http.services.gateway-service.loadbalancer.server.port=8080
+#      - traefik.http.routers.gateway-management-router.tls=true
```

Comment:
Could we not just leave all lines as is and set tls=false?

And just disable traeffic for the containers we do not need it yet instead of comment out/in.

See: https://doc.traefik.io/traefik/providers/docker/#exposedbydefault

Reply:
If possible, can we have the gateway without being hidden behind a proxy? This will degrade the performance a fair bit and it would be nice to demonstrate Apiman in the best light.

Not a major point if it's not possible, but shouldn't be too  hard I guess.